[CI] DynamicMappingIT testDynamicRuntimeObjectFields failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/bo4ybzyqwb6to/tests/:server:internalClusterTest/org.elasticsearch.index.mapper.DynamicMappingIT/testDynamicRuntimeObjectFields

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.index.mapper.DynamicMappingIT.testDynamicRuntimeObjectFields" -Dtests.seed=9D50C5C879C68922 -Dtests.locale=hr-HR -Dtests.timezone=America/Sitka -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.index.mapper.DynamicMappingIT&tests.test=testDynamicRuntimeObjectFields

**Failure excerpt:**
```
junit.framework.AssertionFailedError: Unexpected exception type, expected MapperParsingException but got java.lang.IllegalArgumentException: mapper [obj.runtime.dynamic.number] cannot be changed from type [long] to [text]

  at __randomizedtesting.SeedInfo.seed([9D50C5C879C68922:4B9879BD3FB9E18A]:0)
  at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2879)
  at org.apache.lucene.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2860)
  at org.elasticsearch.index.mapper.DynamicMappingIT.testDynamicRuntimeObjectFields(DynamicMappingIT.java:467)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.IllegalArgumentException: mapper [obj.runtime.dynamic.number] cannot be changed from type [long] to [text]

    at org.elasticsearch.index.mapper.FieldMapper.checkIncomingMergeType(FieldMapper.java:420)
    at org.elasticsearch.index.mapper.FieldMapper.merge(FieldMapper.java:405)
    at org.elasticsearch.index.mapper.FieldMapper.merge(FieldMapper.java:49)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:388)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:348)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:374)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:348)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:374)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:348)
    at org.elasticsearch.index.mapper.ObjectMapper.doMerge(ObjectMapper.java:374)
    at org.elasticsearch.index.mapper.RootObjectMapper.doMerge(RootObjectMapper.java:328)
    at org.elasticsearch.index.mapper.ObjectMapper.merge(ObjectMapper.java:348)
    at org.elasticsearch.index.mapper.RootObjectMapper.merge(RootObjectMapper.java:323)
    at org.elasticsearch.index.mapper.Mapping.merge(Mapping.java:136)
    at org.elasticsearch.index.mapper.MapperService.mergeMappings(MapperService.java:386)
    at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.applyRequest(MetadataMappingService.java:115)
    at org.elasticsearch.cluster.metadata.MetadataMappingService$PutMappingExecutor.execute(MetadataMappingService.java:79)
    at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:828)
    at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:395)
    at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:235)
    at org.elasticsearch.cluster.service.MasterService$Batcher.run(MasterService.java:167)
    at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:146)
    at org.elasticsearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:202)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:712)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:262)
    at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:225)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/index/mapper/DynamicMappingIT.java
@@ -447,10 +447,13 @@ public class DynamicMappingIT extends ESIntegTestCase {
         );
 
         // the parent object has been mapped dynamic:true, hence the field gets indexed
+        // we use a fixed doc id here to make sure this document and the one we sent later with a conflicting type
+        // target the same shard where we are sure the mapping update has been applied
         assertEquals(
             RestStatus.CREATED,
             client().prepareIndex("test")
                 .setSource("obj.runtime.dynamic.number", 1)
+                .setId("id")
                 .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
                 .get()
                 .status()

